### DISCUSSION TITLE
Why not optimize images and even change extension to webp ?

### DISCUSSION THREAD

**Discussion by hamidovz** [kodadot/nft-gallery#10019] [General]
hi all ! i am new here. i saw that the website's performance super slow. when i profiled it i realized that it has a lot of heavy images. i don't have enough info about this project. But i am curious why not to optimize images or even use webp extension ? is that has to do with nft system ? if optimizing images is allowed and image uploading codes written in js , then i can optimize them. this will make website much more faster.

for example this image is almost 10mb and this is way too big for a thumbnail : **https://image.w.kodadot.xyz/ipfs/bafybeibkvuniddxpvsazvzzqpmqsqjrlqa55y7ssuea3ic3zyv2bstf5oi/0.png**

i saw a lot of similar big sized images especially in this section:

![image](https://github.com/kodadot/nft-gallery/assets/38864838/1aaee385-6728-46d7-b255-002f1c0387b5)

**Comment by exezbcz** [comment]
cc @kodadot/internal-dev 👀

**Comment by vikiival** [comment]
> when i profiled it i realized that it has a lot of heavy images. 

It is very strange behaviour as we use [Cloudflare Images](https://www.cloudflare.com/en-gb/developer-platform/cloudflare-images/) to make responses faster

> 

> if optimizing images is allowed and image uploading codes written in js , then i can optimize them. this will make website much more faster.

[You can take a look at the code](https://github.com/kodadot/workers/tree/main/image)

**Comment by preschian** [comment]
Hi, thanks for raising the issue

Will check. For images, it should redirect to cf-images with this format:
- https://image.w.kodadot.xyz/ipfs/bafybeibkvuniddxpvsazvzzqpmqsqjrlqa55y7ssuea3ic3zyv2bstf5oi/0.png. should be redirected to ->
- https://imagedelivery.net/jk5b6spi_m_-9qC4VTnjpg/bafybeibkvuniddxpvsazvzzqpmqsqjrlqa55y7ssuea3ic3zyv2bstf5oi/0.png/public

Related to webp, Cloudflare images will auto-serve as webp/avif depending on browser support

**Comment by preschian** [comment]
fixed by https://github.com/kodadot/workers/pull/279
